smslibrary/src/main/java/com/eis/smslibrary/SMSSentBroadcastReceiver.java

```diff
@@ -0,0 +1,78 @@
+package com.eis.smslibrary;
+
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.telephony.SmsManager;
+
+import androidx.annotation.NonNull;
+
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+/**
+ * Broadcast receiver for sent messages, called by Android Library.
+ * Must be instantiated and set as receiver with context.registerReceiver(...)
+ */
+class SMSSentBroadcastReceiver extends BroadcastReceiver {
+
+    private SMSSentListener listener;
+    private SMSMessage message;
+
+    /**
+     * Constructor for the custom {@link BroadcastReceiver}.
+     *
+     * @param message  message that will be sent.
+     * @param listener listener to be called when the operation is completed successfully or not.
+     */
+    SMSSentBroadcastReceiver(@NonNull final SMSMessage message,@NonNull final SMSSentListener listener) {
+        this.listener = listener;
+        this.message = message;
+    }
+
+    /**
+     * @param listener called when the message is sent.
+     */
+    void setListener(@NonNull final SMSSentListener listener) {
+        this.listener = listener;
+    }
+
+    /**
+     * @param message passed to the listener when it is sent.
+     */
+    void setMessage(@NonNull final SMSMessage message) {
+        this.message = message;
+    }
+
+    /**
+     * This method is subscribed to the intent of a message sent, and will be called whenever a message is sent using this library.
+     */
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        SMSMessage.SentState sentState = SMSMessage.SentState.ERROR_GENERIC_FAILURE;
+
+        switch (getResultCode()) {
+            case Activity.RESULT_OK:
+                sentState = SMSMessage.SentState.MESSAGE_SENT;
+                break;
+            case SmsManager.RESULT_ERROR_RADIO_OFF:
+                sentState = SMSMessage.SentState.ERROR_RADIO_OFF;
+                break;
+            case SmsManager.RESULT_ERROR_NULL_PDU:
+                sentState = SMSMessage.SentState.ERROR_NULL_PDU;
+                break;
+            case SmsManager.RESULT_ERROR_NO_SERVICE:
+                sentState = SMSMessage.SentState.ERROR_NO_SERVICE;
+                break;
+            case SmsManager.RESULT_ERROR_LIMIT_EXCEEDED:
+                sentState = SMSMessage.SentState.ERROR_LIMIT_EXCEEDED;
+                break;
+        }
+
+        if (listener != null) //extra check, even though listener should never be null
+            listener.onSMSSent(message, sentState);
+
+        context.unregisterReceiver(this);
```

Comment:
What do you mean with "parent method", this method is inherited by BroadcastReceiver. I think it makes no difference unregistering the receiver here or at the top of the method, it just seemed reasonable to do it at the end of all the processing.
Also, what do you mean by registered again somewhere else? Why would it be registered again? There is will be one different BroadcastReceiver per message, so it doesn't need to be registered again.

Reply:
Probably the methods `setListener` and `setMessage` makes it confusing because they are actually useless, you can't change listener once you've sent the message and you're waiting for the callback

Reply:
Ok then, i thought that it was a receiver for every message and when it would be unregistered it wouldn’t work again

Reply:
> Probably the methods `setListener` and `setMessage` makes it confusing because they are actually useless, you can't change listener once you've sent the message and you're waiting for the callback

Yes, this was a little bit confusing, i suggest you to leave a comment on these lines or remove them if they are useless

Reply:
Will remove them